apps/andi/lib/andi/harvest/harvester.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Andi.Harvest.Harvester do
+  @moduledoc """
+  retreives data json and maps it to a smartcity dataset
+  """
+  use Tesla
+
+  import Andi
+  import SmartCity.Event, only: [dataset_update: 0]
+
+  alias Andi.Harvest.DataJsonToDataset
+
+  require Logger
+
+  def start_harvesting(org) do
+    url = org.dataJsonUrl
+
+    with {:ok, data_json} <- get_data_json(url),
+         {:ok, decoded_data_json} <- Jason.decode(data_json),
+         datasets <- map_data_json_to_dataset(decoded_data_json, org),
+         :ok <- dataset_update(datasets) do
+      :ok
+    else
+      error ->
+        {:error, error}
+    end
+  end
+
+  def get_data_json(url) do
+    case get(url) do
+      {:ok, response} ->
+        {:ok, response.body}
+
+      {:error, error} ->
+        Logger.error("Failed to get data json from #{url}: #{inspect(error)}")
+        {:error, error}
+    end
+  end
+
+  def map_data_json_to_dataset(data_json, org) do
+    DataJsonToDataset.mapper(data_json, org)
```

Comment:
It will probably make sense to pull this into a shared library with the dataset -> data.json mapping we use in discovery, but that's another effort.